src/parquet/encodings/plain-encoding.h

```diff
@@ -52,6 +52,19 @@ class PlainDecoder : public Decoder {
     return GetValues(buffer, max_values, sizeof(int64_t));
   }
 
+  virtual int GetInt96(Int96* buffer, int max_values) {
+    max_values = std::min(max_values, num_values_);
+    int size = 3*sizeof(uint32_t);//Struct Int96 could be padded
```

Comment:
The code from flatbuffers looks reasonable to me. 
However, I am inclined towards having the padding to allow aligned memory access. Can you explain where the padding is an issue in a template context ? If required, we can also specialize the template for "Int96" type at places where there is a problem.